server/src/controller/supplier.ts

```diff
@@ -0,0 +1,136 @@
+import { QueryRunner, Repository } from "typeorm";
+/* eslint-disable import/prefer-default-export */
+import {
+  ErrorType,
+  SuccessType,
+  SupplierCreate,
+  SupplierResponse,
+  SupplierUpdate,
+} from "../types";
+import AppDataSource from "../db";
+import { Supplier } from "../model";
+
+const queryRunner: QueryRunner = AppDataSource.createQueryRunner();
+const supplierRepository: Repository<Supplier> =
+  AppDataSource.getRepository(Supplier);
+
+export const createSupplier = async (
+  body: SupplierCreate,
+): Promise<ErrorType | SuccessType> => {
+  try {
+    await queryRunner.startTransaction();
+
+    const supplier = new Supplier();
+    supplier.name = body.name;
+    supplier.phone = body.phone;
+    supplier.address = body.address;
+    supplier.email = body.email;
+
+    await queryRunner.manager.save(supplier);
+
+    await queryRunner.commitTransaction();
+
+    return {
+      data: {
+        id: supplier.id,
+        name: supplier.name,
+        email: supplier.email,
+        phone: supplier.phone,
+        address: supplier.address,
+      },
+    };
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (err: any) {
+    await queryRunner.rollbackTransaction();
+    if (err.code === "23505") {
+      return {
+        errorKey: "email",
+        errorDescription: err.detail,
+        errorCode: 409,
+      };
+    }
+    return {
+      errorKey: "unknown",
+      errorDescription: "unknown error",
+      errorCode: 500,
+    };
+  }
+};
+
+export const getAllSuppliers = async (): Promise<SupplierResponse[]> => {
+  const allSuppliers: SupplierResponse[] = await supplierRepository
+    .createQueryBuilder("supplier")
+    .select("supplier.id")
+    .addSelect("supplier.name")
+    .addSelect("supplier.email")
+    .addSelect("supplier.phone")
+    .addSelect("supplier.address")
+    .orderBy("supplier.name")
+    .getMany();
+
+  return allSuppliers;
+};
+
+export const getOneSupplier = async (
```

Comment:
In order the put request to work, you will need to return the supplier id as part of the getOneSupplier